### ISSUE TITLE
Speculative Decoding

### ISSUE THREAD

**Issue by RohitMidha23** [SYSTRAN/faster-whisper#771]
1. Is speculative decoding faster than faster-whisper? 

2. Is there going to be support anytime soon for speculative decoding in faster-whisper?

Both of these questions are asked with a purely realtime, streaming audio view in mind!

**Comment by trungkienbkhn** [comment]
@RohitMidha23 , hello.

1. I tested speculative decoding for whisper from [here](https://github.com/openai/whisper/discussions/1914), then compared with fw large-v2. I used an audio file 192s and RTX 3090 GPU card. Below are the transcription times:

+ fw large-v2: 186.488s
+ HF speculative decoding: 173.939s
=> It can be seen that speculative decoding support from HF transformers has improved the time a bit, but not too much.

2. Yes we will look into this
